python/pyspark/_globals.py

```diff
@@ -59,10 +59,6 @@ def __new__(cls):
             cls.__instance = super(_NoValueType, cls).__new__(cls)
         return cls.__instance
 
-    # needed for python 2 to preserve identity through a pickle
-    def __reduce__(self):
-        return (self.__class__, ())
-
```

Comment:
This one I feel less strongly but I didnt touch because this class was directly ported from NumPy.

Reply:
Thanks for clearing up, lets keep it for now.